source/library/CabalGild/Unstable/Action/EvaluatePragmas.hs

```diff
@@ -76,21 +76,17 @@ discover p n fls dt ds = do
   let (excs, incs) = Either.partitionEithers flgs
   mapM_ (Exception.throwM . UnknownOption.fromString) opts
   mapM_ (Exception.throwM . InvalidOption.fromString) errs
-  let root = FilePath.takeDirectory p
+  let root = FilePath.dropTrailingPathSeparator . normalize $ FilePath.takeDirectory p
       directories =
         List.nubOrd
-          . fmap
-            ( FilePath.dropTrailingPathSeparator
-                . normalize
-                . FilePath.combine root
-            )
+          . fmap normalize
           $ if null args then ["."] else args
-  let exclusions = List.nubOrd $ fmap (normalize . FilePath.combine root) excs
+      exclusions = List.nubOrd $ fmap normalize excs
       inclusions =
         List.nubOrd
-          . fmap (normalize . FilePath.combine root)
+          . fmap normalize
           $ if null incs then ["**"] else incs
-  files <- Trans.lift $ MonadWalk.walk "." inclusions exclusions
+  files <- Trans.lift $ MonadWalk.walk root inclusions exclusions
```

Comment:
The basic idea of this fix is that we'll use `root` rather than the current directory to search for files. The files returned are relative to the `root`, and the inclusions and exclusions are also evaluated relative to the `root`. So everything works out nicely without having to do too much. 

Removing a lot of calls to `combine root` is a nice bonus.